src/shared/components/query/QueryStore.ts

```diff
@@ -592,15 +607,71 @@ export class QueryStore
 		}
 	});
 
+	readonly profiledSamplesCount = remoteData<{ mutation: number, cna: number, mutation_cna: number }>({
+		await: () => [this.sampleListInSelectedStudies],
+		invoke: async () => {
+			const profiledSamples = await Promise.resolve(categorizedSamples(this.sampleListInSelectedStudies.result!));

```

Comment:
it seems categorizedSamples is a synchronous function, so why do we need to wrap in a promise and then await it?

Reply:
true, its not rquired. Forgot to refactor when changing from LazyMobXCache to MobxPromiseCache